### ISSUE TITLE
Disabling glow & halo

### ISSUE THREAD

**Issue by LSAXYZ321** [Stellarium/stellarium#3159]
I can't find any option for permanent disabling glow & halo (no matter zoom level), in order to see planets and moons realistic
In the example shown with Neptune and its moons, they simply look wrong and way too large because of the glow
Furthermore, it seems there is a limitation for zoom, since FOV can't get smaller than 0,00139° as shown in the image
If I zoom out the moon Triton appears almost as big as Neptune because of the glow
Would be good if the zoom factor could be without limit, or at least much better
Stellarium-web can zoom 5 times more down to af FOV of 0,000278
Is it possible in Stellarium?, if not consider this a feature request

BR Lars

![ST_p01](https://user-images.githubusercontent.com/95540897/229564856-fc6657df-6f4c-4ba1-a72f-0775506a7db1.png)

**Comment by LSAXYZ321** [comment]
gzotti: "has help for you. Spoiler: Section D.1.15"

Thanks, found config.ini in
c:\Users\[username]\AppData\Roaming\Stellarium\
, and added the line
min_fov = 0.0000015

gzotti: "The glow cannot be disabled, else too small spheres would vanish entirely."

So it is now a feature request to be able to disable glow and halo, so Stellarium can show a realistic night sky, where moons and planets might be a faint pixel or vanishing completely
Hope this can be made for better realism
As far as I know Starry Night can do that correct, but for a prize

gzotti: "You can scale up planets and moons."

Tried that feature, doesn't make sense to "grow" a moon ...

**Comment by Atque** [comment]
I think more scaling options for halos would solve this. Starry Night has 4 or 5 if I remember correctly.

**Comment by 10110111** [comment]
Ideally, this should be rendered in HDR in a physically-based manner, and then bloom effect should be applied to make haloes with correct sizes and brightnesses.

**Comment by LSAXYZ321** [comment]
I just updated to Stellarium 23.1 and see that this graphical issue isn't solved, in a way so we can see a realistic night sky without this glow for planets and moons
This attached screendump show planets like Venus, Mercury and Jupiter more glowing than the sun ....
Would be great with an option to disable the glow effect, or in case anyone like that, it could a glow option from 0 - 100 %
![ST_p03](https://user-images.githubusercontent.com/95540897/233848370-98c04239-fa70-4dae-affc-c3936dd30864.png)

**Comment by gzotti** [comment]
The halos try to represent the solar system objects in the same way as stars. Switching them off means invisible spheres (zero size, zero brightness!) Of course, if you switch off the special large halo of the sun, this is what you get. The sun is large enough to remain visible.

**Comment by alex-w** [comment]
@LSAXYZ321 did you tried set `halo=false` for planets in `ssystem.ini` file?

**Comment by gzotti** [comment]
Yes, a good demostration of the bad idea.

**Comment by LSAXYZ321** [comment]
gzotti: "The halos try to represent the solar system objects in the same way as stars"

This is just scientifically wrong, since halo's might only appear for bright objects as the sun and the moon, under certain atmospheric conditions, and never for other planets and moons
I might even describe this glow and halo as some kind of display error, not a feature

gzotti: "Switching them off means invisible spheres (zero size, zero brightness!)"

That would also be wrong, since its not zero size/ brightness, but just very small hence not visible without zooming in with a smaller FOV
It's the same in the real night sky since distant planets and certainly their moons will be invisible to the naked eye, it should be possible to get the same display in Stellarium
We can still enable planet labels as well as we can search for celestial objects via F3

@alex-w I will try the halo=false setting and see how that works

**Comment by Atque** [comment]
The difference in brightness between e.g. Venus and Mercury needs to be visualized in large FOV's. The only way to do that, is to draw their halos at different sizes. And if you see Venus and Mercury in the sky beside each other, the former will appear larger than the latter.

**Comment by 10110111** [comment]
> This is just scientifically wrong, since halo's might only appear for bright objects as the sun and the moon

If we start saying about science, we must admit that there's no principal difference between e.g. the Moon and Venus. Both can result in halos, but the halo for the Moon is several orders of magnitudes brighter than that of Venus.

> under certain atmospheric conditions

The thing called halo in atmospheric optics is not what's rendered in Stellarium. Stellarium attempts to draw the combination of glare and some geometric aberrations of the optical system of human eyes.

> I might even describe this glow and halo as some kind of display error, not a feature

No, the intention was to make it closer to what a human would see, it's not a display error. Though, the results, despite being somewhat representative in some cases, still leave a lot to be desired in general.

**Comment by gzotti** [comment]
> gzotti: "The halos try to represent the solar system objects in the same way as stars"
> 
> This is just scientifically wrong, since halo's might only appear for bright objects as the sun and the moon, under certain atmospheric conditions, and never for other planets and moons I might even describe this glow and halo as some kind of display error, not a feature

I know this is not a halo as in atmospheric optics.  The soft glow which looks just any star and is used to represent point-like objects has been named "halo" in this program for probably the last 21 years or so. 

> gzotti: "Switching them off means invisible spheres (zero size, zero brightness!)"
> 
> That would also be wrong, since its not zero size/ brightness, but just very small hence not visible without zooming in with a smaller FOV It's the same in the real night sky since distant planets and certainly their moons will be invisible to the naked eye, it should be possible to get the same display in Stellarium We can still enable planet labels as well as we can search for celestial objects via F3

A 0.01 pixel wide sphere is not rendered. Also a 1-size white pixel is a bad representation for Jupiter when compared to mag -1 stars. As long as HDR displays are not commonplace and HDR rendering with auto-bloom has not been implemented, we have to be happy with a white disk of magnitude-dependent size. But @LSAXYZ321 certainly knows better how to solve that topic.

**Comment by LSAXYZ321** [comment]
Atque: "The difference in brightness between e.g. Venus and Mercury needs to be visualized in large FOV's. The only way to do that, is to draw their halos at different sizes"

Brightness can be applied by RGB / HSL values only, also for 8-bit gradient base

Atque: "And if you see Venus and Mercury in the sky beside each other, the former will appear larger than the latter"

And so it should be shown

10110111: "we must admit that there's no principal difference between e.g. the Moon and Venus"

Correct

10110111: "The thing called halo in atmospheric optics is not what's rendered in Stellarium"

I'm aware of that

gzotti: "A 0.01 pixel wide sphere is not rendered."

Correct and so it should be: Not rendered

gzotti: "Also a 1-size white pixel is a bad representation for Jupiter when compared to mag -1 stars"

Obviously, it shouldn't be a white pixel, but dimmed to the merged color of Jupiter

gzotti: "As long as HDR displays are not commonplace"

HDR isn't that necessary, as mentioned earlier
Brightness can be applied by RGB / HSL values only, also for 8-bit gradient base

gzotti: "we have to be happy with a white disk of magnitude-dependent size"

The unrealistic way Stellarium represents planets now are quite bad..

**Comment by gzotti** [comment]
Saying "not like this" is easy. We are still waiting for your complete solution. You can test your proposal as described above.

**Comment by LSAXYZ321** [comment]
gzotti: "We are still waiting for your complete solution"

Just look at this video from Stellarium
From the start, the Mars moon Deimos are shown correct, with Gouraud shading, or similar to
When zooming out, it should continue with the same algorithm, to a grayish pixel fading away, quite simple

But at 0:07 glow start on Deimos, unfortunately

At 0:15 we see Mars shaded correctly, and for some time while zooming out we see the color blending, like downsizing an image in a graphics 

https://user-images.githubusercontent.com/95540897/234405205-86333c6b-dd1d-4e62-a020-353993ee331d.mp4

program.
Here. it's just a natural result of Gouraud shading, which you will know if you have worked in the field of 3D-CAD and rendering
At 0:24 glow start on Mars, later on still while zooming out, Mars even "swallows" a bunch of stars

It's evident that the glow comes from another algorithm, not Gouraud shading and I'm just asking for an option to disable that
So for the most likely very few users who by some strange reasons do like this glow, you can enable it

**Comment by LSAXYZ321** [comment]
gzotti: "We are still waiting for your complete solution"

Just look at this video from Stellarium
From the start, the Mars moon Deimos are shown correct, with Gouraud shading, or similar to
When zooming out, it should continue with the same algorithm, to a grayish pixel fading away, quite simple

But at 0:07 glow start on Deimos, unfortunately

At 0:15 we see Mars shaded correctly, and for some time while zooming out we see the color blending, like downsizing an image in a graphics-program.
Here. it's just a natural result of Gouraud shading, which you will know if you have worked in the field of 3D-CAD and rendering
At 0:24 glow start on Mars, later on still while zooming out, Mars even "swallows" a bunch of stars

It's evident that the glow comes from another algorithm, not Gouraud shading and I'm just asking for an option to disable that
So for the most likely very few users who by some strange reasons do like this glow, you can enable it


https://user-images.githubusercontent.com/95540897/234405642-813d7de7-9902-4dae-9a0e-b96966a9b65a.mp4

**Comment by gzotti** [comment]
You can test your proposal as described above.

**Comment by LSAXYZ321** [comment]
@gzotti as mentioned watch the video and read the explanation, quite simple ....

**Comment by chanokim** [comment]
this issue makes planets and stars look ridiculously larger than real especially in lower magnification telescope. It is far from realistic and renders Stellarium less usable, especially on double stars.

**Comment by gzotti** [comment]
So? You can finetune star scales in the GUI to your liking. Or suggest working code which is an improvement in all cases.
Else: there is always other software out there.

**Comment by gzotti** [comment]
You can test your proposal as described above.

**Comment by LSAXYZ321** [comment]
chanokim: "It is far from realistic and renders Stellarium less usable, especially on double stars"

Agree, horrible graphics ..

gzotti: "Or suggest working code which is an improvement in all cases"

The code is already in Stellarium as I pointed out in april
The simple solution is an option to disable the activation of the "glow" algorithm as demonstrated in the video
I'm sure that those involved in the development can correct this, since its a simple job
If not, depending on code language, I can do that, but not for free ...

gzotti: "there is always other software out there"

Correct, another free option, although a "space simulation" not astronomy, is the older Celestia
Celestia — real-time 3D visualization of space
https://celestiaproject.space/

For astronomy software and way better than Stellarium, there is 
Starry Night - Professional Astronomy Telescope Control Software
https://starrynight.com/starry-night-8-professional-astronomy-telescope-control-software.html
, although not free
When I realised that the graphics "bug" in Stellarium 23.2 isn't solved i'm considering buying Starry Night

**Comment by 10110111** [comment]
> its a simple job

Stellarium has to make trade-offs to get a balance between two conflicting goals: to show relative visibility of objects and make the look more or less realistic. There's no way to make, say, Venus much brighter than, say, Vega other than by making the dot larger, which then conflicts with the apparent size of the Moon. On HDR there's a better way, but Stellarium doesn't support this yet.

> If not, depending on code language, I can do that, but not for free
> ...
> i'm considering buying Starry Night

That's an interesting switch from almost asking for a payment to considering a spending ;)

Seriously, don't forget that Stellarium is free (in both senses) and developed by enthusiasts in their free time. If you want something to be done, you can actually do it yourself—or wait until someone does it for you (possibly after a long time). But what's supposed to be merged upstream needs to be compatible with all the goals of the project.

Actually, a more realistic physically-based rendering of the whole scene is on my TODO list, and I have some ideas on how to implement it, so at some point we may get something. But again, there's no guarantee on how soon this will be completed, because I also do it in my free time.

**Comment by gzotti** [comment]
We have described above what YOU can do _on your private installations_ to make objects vanish when their screen size becomes too small. For now, we find that a bad idea.

**Comment by LSAXYZ321** [comment]
10110111: "Stellarium has to make trade-offs to get a balance between two conflicting goals: to show relative visibility of objects and make the look more or less realistic."

As mentioned, as you could simply add an option for users to disable the "glow" algorithm

10110111: "That's an interesting switch from almost asking for a payment to considering a spending ;)"

For a start you didn't wonder why gzotti, asked a user (me) for a code-proposal and on which basis ?
His post's didn't bring any positive contributions to the topic

10110111: "Seriously, don't forget that Stellarium is free (in both senses) and developed by enthusiasts in their free time"

I'm aware and in a community like this, it's also normal that users come with ideas/ suggestions for the software
I hope its ok here also ??

10110111: "If you want something to be done, you can actually do it yourself"

Correct, hence my former reply as you quoted

**Comment by 10110111** [comment]
> For a start you didn't wonder why gzotti, asked a user (me) for a code-proposal and on which basis ?

Oh, this is obvious: the development from users' side here happens via Pull Requests, as in any open-source project on GitHub (and generally via patch sets in projects on other platforms). The contributors don't usually expect to be paid when proposing their code, because the merging of their patches is in their best interest.

> it's also normal that users come with ideas/ suggestions for the software
> I hope its ok here also ??

It's OK. It's also OK for the developers to not be interested in a feature unless it's proposed in a pull request that implements it (without breaking other functionality). Again, this is how any opensource project works. You can even fork Stellarium to implement your feature without merging it back upstream if you don't want to publish your code without being paid, it's your freedom (limited only by the GPL).

**Comment by LSAXYZ321** [comment]
10110111: "Oh, this is obvious: the development from users' side here happens via Pull Requests"

Ok, so where do you see any "pull request" from me?
This is an "issues" group, just to clarify ;-)

10110111: "It's also OK for the developers to not be interested in a feature"

Of course, in the same way as users are free to look for better alternatives ...
Besides that, this debate does not contribute to anything positive related to the described issue ....

**Comment by 10110111** [comment]
> Ok, so where do you see any "pull request" from me?

This is the point! @gzotti asked you for a code proposal, which you refuse to provide. If nobody wants to do something, it won't be done.

**Comment by LSAXYZ321** [comment]
10110111: "This is the point! @gzotti asked you for a code proposal, which you refuse to provide"

How is it that you ang gzotti conclude that a all users, daring to come with a suggestion, is also a developer, willing to work for free, with the risk that the work isn't approved, hence "proposal" ?

btw: The code for correct visualisation is already in Stellarium, as I explained and demonstrated in the video, hence the use of 2 different algorithm, one without glow, another with the glow "feature"
The only thing missing is an option in the configuration
There is no point in further debating, since you made it clear that an option for disabling the glow feature, wouldn't be done

**Comment by 10110111** [comment]
> How is it that you ang gzotti conclude that a all users, daring to come with a suggestion, is also a developer, willing to work for free, with the risk that the work isn't approved, hence "proposal" ?

We don't. It's just that if you don't like the suggestions that were provided here, it's up to you to come up with a solution that would satisfy you. Yes, you might not be a developer, as most of the users, but then you can't demand that something is implemented just because you don't like what's available. If a developer appears that sympathizes with your idea, they may be able to implement it. But until then, if you're not a developer and still badly want this done, just hire one—or, as you "threaten", find better software.

**Comment by gzotti** [comment]
Why don't you just apply the solution we have suggested above? After many years of repeatedly trying to fine-tune the appearance of a star-like replacement "dot" when the 3D body becomes too small to render, we conclude that it is a bad idea to allow switching off the feature that has been called "halo" (without implying this is the halo of atmospheric optics) in the early years of the project. Doing that will erase the planets from the sky overview. If you do not believe it, just try yourself. We won't waste more time on it, and therefore decided to close this issue. 

I see the next change in code required whenever HDR rendering/filtering/autobloom becomes reality.

**Comment by Chris2000SP** [comment]
Ahm i wasn't aktive here at 2023 when this discussion happened but i noticed that i could not use Stellarium with my Settings of my Teleskope and Eye peaces. Example Mak 127/1900 (Bresser) and Eye peace 10mm (Baader Hyperion) or the Eye peaces i have from the Set i bought (26mm, 10mm). For that matter i used Stellarium to track Moons of Planets but with that glow and the settings i use i can not keep this apart anymore. And of Saturn i do not see the Ring anymore with this Setting.

I did not read this discussion and i won't read it because i noticed some serious dispute talking here and i have a reason to not get slipping into mental health problems because of that. But i want to mention that i know how i see the Saturn in my Mak and the glow effect is not the right way.

Cheers

**Comment by LSAXYZ321** [comment]
@Chris2000SP If you look for planetarium software with a realistic representation of the nightsky (without unrealistic glow "feature"), there might be some options like "Starry Night 8", although not for free ...
[www.starrynight.com](url)

For solarsystem simulation there is also Celestia, which is for free and also with realistic representation
However this can't control a telescope
[www.celestiaproject.space](url)

If you find other better alternatives than Stellarium, please let me know
